cluster/swarm/cluster.go

```diff
@@ -511,21 +511,38 @@ func (c *Cluster) CreateVolume(request *types.VolumeCreateRequest) (*types.Volum
 	}
 	if node == "" {
 		c.RLock()
-		for _, e := range c.engines {
-			wg.Add(1)
-
-			go func(engine *cluster.Engine) {
-				defer wg.Done()
 
-				v, er := engine.CreateVolume(request)
+		// When the scope is `local` we need to create the volume everywhere
+		var i int
+		for _, e := range c.engines {
+			if i == 0 {
+				v, er := e.CreateVolume(request)
 				if v != nil {
 					volume = v
 					err = nil
+					if v.Scope == "global" {
```

Comment:
If CreateVolume fails for whatever reason, `v` is `nil` and this would not be checked. It'd end up creating Volume on other engines. 

Can scope be derived from request directly? So that it's always clear if it's `global` or `local`.